test/images/agnhost/openidmetadata/openidmetadata.go

```diff
@@ -162,3 +172,32 @@ func withOAuth2Client(context.Context) (context.Context, error) {
 		})
 	return ctx, nil
 }
+
+// DNS can be available sometime after the container starts due to the way
+// networking is set up for Windows nodes with dockershim as the container runtime.
+// In this case the. we should make sure we are able to resolve the issuer before
+// invoking oidc.NewProvider.
+// See https://github.com/kubernetes/kubernetes/issues/99470 for more details.
+func ensureWindowsDNSAvailability(issuer string) error {
+	log.Println("Ensuring Windows DNS availability")
+
+	u, err := url.Parse(issuer)
+	if err != nil {
+		return err
+	}
+
+	for retry, timeout := 0, []time.Duration{1 * time.Second, 2 * time.Second}; ; retry++ {
```

Comment:
Can we use `k8s.io/apimachinery/pkg/util/wait` for the wait/timeout logic? `wait.PollImmediate` is probably a simple solution here.